File: CSES/2. Sorting and Searching/CSES-1084_Apartments.cpp
#include <bits/stdc++.h>
using namespace std;
using LL = long long;
const int INF = 0x3f3f3f3f;
const int MOD = 1e9 + 7;
const int N = 100005;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int n, m, k;
    cin >> n >> m >> k;
    vector<int> arr(n), brr(m);
    for (int i = 0; i < n; i++) cin >> arr[i];
    for (int i = 0; i < m; i++) cin >> brr[i];
    sort(arr.begin(), arr.end());
    sort(brr.begin(), brr.end());
    int ans = 0;
    int i = 0, j = 0;
    while (i < n && j < m) {
        if (arr[i] + k < brr[j]) i++;
        else if (arr[i] - k > brr[j]) j++;
        else i++, j++, ans++;
    }
    cout << ans << endl;
    return 0;
}

File: CSES/2. Sorting and Searching/CSES-1084_Apartments.py
n, m, k = map(int, input().split())
arr = sorted(map(int, input().split()))
brr = sorted(map(int, input().split()))

ans = 0
i = j = 0
while i < n and j < m:
    if arr[i] + k < brr[j]: # 公寓面積比最大需求大，換下一個人
        i += 1
    elif arr[i] - k > brr[j]: # 公寓面積比最小需求小，換下一個公寓
        j += 1
    else: # 符合條件
        i += 1
        j += 1
        ans += 1
print(ans)